dpnp/dpnp_iface_statistics.py

```diff
@@ -104,31 +105,40 @@ def mean(a, axis=None):
 
     """
 
-    if use_origin_backend():
-        return numpy.mean(a, axis=axis)
+    dim_input = input.ndim
 
-    # TODO This block needs to be redesigned
-    # Test pytest tests/third_party/cupy/statistics_tests/test_mean.py::TestMeanVar::test_mean_all
-    # failed if uncomment the code
-    # if numpy.shape(a) == ():
-    #     if axis is None:
-    #         return a
-    #     else:
-    #         raise AttributeError("Object has no attribute 'mean'")
+    if dim_input == 0:
+        """
+        Means scalar at input
+        input.shape[0] will failed because it is list
+        TODO need to copy it into new 'result' array
+        """
+        return numpy.mean(input, axis=axis)
 
-    if not isinstance(axis, int) and axis is not None:
-        raise TypeError("Parameter axis can be only int!")
+    if dim_input > 2 and axis is not None:
+        raise NotImplementedError
 
-    if isinstance(a, dparray):
-        result = dpnp_mean(a, axis=axis)
+    is_input_dparray = isinstance(input, dparray)
+
+    if axis is not None and (axis >= dim_input or -1 * axis >=dim_input):
+        return numpy.mean(input, axis=axis)
+
+    if not use_origin_backend(input) and is_input_dparray:
```

Comment:
it would be better to combine these 3 checks into one.
on lines 110, 123 and 126

Reply:
Thank you!
Is it possible to combine last two (112 and 116 lines) into one?
as a result we will have only one `numpy.mean` call in this fucntion.

Reply:
@shssf  112 call numpy.mean, 116 - dpnp_mean

Reply:
yes, and 131 calls numpy.mean also :-)
anyway, let's merge this because it is already looks much better :-)